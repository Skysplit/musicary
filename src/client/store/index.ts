import { createStore } from 'redux';
import { StoreCreatorOptions } from 'next-redux-wrapper';
import { Request } from 'express';
import rootReducer from './rootReducer';
import middleware from './middleware';

export type State = ReturnType<typeof rootReducer>;

interface Options extends StoreCreatorOptions<State, {}, {}, {}, {}> {
  req: Request;
}

export default (initialState: State, options: Options) => {
  return createStore(rootReducer, middleware);
};
