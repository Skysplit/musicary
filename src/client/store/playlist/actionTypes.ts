export const FETCH_MANY_PLAYLISTS_REQUEST = 'playlist/FETCH_MANY_REQUEST';
export const FETCH_MANY_PLAYLISTS_SUCCESS = 'playlist/FETCH_MANY_SUCCESS';
export const FETCH_MANY_PLAYLISTS_FAILURE = 'playlist/FETCH_MANY_FAILURE';

export const FETCH_SINGLE_PLAYLIST_REQUEST = 'playlist/FETCH_SINGLE_REQUEST';
export const FETCH_SINGLE_PLAYLIST_SUCCESS = 'playlist/FETCH_SINGLE_SUCCESS';
export const FETCH_SINGLE_PLAYLIST_FAILURE = 'playlist/FETCH_SINGLE_FAILURE';

export const REMOVE_PLAYLIST_REQUEST = 'playlist/REMOVE_REQUEST';
export const REMOVE_PLAYLIST_SUCCESS = 'playlist/REMOVE_SUCCESS';
export const REMOVE_PLAYLIST_FAILURE = 'playlist/REMOVE_FAILURE';

export const SET_PLAYLIST = 'playlist/SET';
