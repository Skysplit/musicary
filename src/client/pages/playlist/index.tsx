import React, { PureComponent } from 'react';
import { Request } from 'express';
import { bindActionCreators } from 'redux';
import { connect, Dispatch } from 'react-redux';
import Error from 'next/error';
import { Context } from 'next/document';
import PlaylistsListContainer from '@client/containers/PlaylistsListContainer';
import client, { getHeaders } from '@client/utils/client';
import { PlaylistInterface } from '@client/store/playlist';
import { fetchPlaylistsSuccess } from '@client/store/playlist/actions';

interface PropsContext extends Context {
  req: Request;
}

interface InitialProps {
  title: string;
  errorCode?: number;
  playlists?: PlaylistInterface[];
}

interface PlaylistsPageProps extends InitialProps {
  fetchPlaylistsSuccess: typeof fetchPlaylistsSuccess;
}

export class PlaylistsPage extends PureComponent<PlaylistsPageProps> {
  static async fetchPlaylists(req: Request) {
    const headers = getHeaders(req);
    const response = await client.get('/playlists', { headers });
    return response.data as PlaylistInterface[];
  }

  static async getInitialProps(ctx: PropsContext): Promise<InitialProps> {
    let errorCode: number;
    let playlists: PlaylistInterface[];

    try {
      playlists = await PlaylistsPage.fetchPlaylists(ctx.req);
    } catch (err) {
      errorCode = err.response.status;
    }

    return {
      errorCode,
      playlists,
      title: 'My playlists',
    };
  }

  constructor(props: PlaylistsPageProps) {
    super(props);

    if (props.playlists) {
      props.fetchPlaylistsSuccess(props.playlists);
    }
  }

  render() {
    const { errorCode } = this.props;

    if (errorCode) {
      return <Error statusCode={errorCode} />;
    }

    return (
      <PlaylistsListContainer />
    );
  }
}

const actionCreators = {
  fetchPlaylistsSuccess,
};

const mapDispatchToProps = (dispatch: Dispatch) => bindActionCreators(actionCreators, dispatch);

export default connect(null, mapDispatchToProps)(PlaylistsPage);
