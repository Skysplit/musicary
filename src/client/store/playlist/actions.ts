import {
  FETCH_MANY_PLAYLISTS_REQUEST,
  FETCH_MANY_PLAYLISTS_SUCCESS,
  FETCH_SINGLE_PLAYLIST_REQUEST,
  FETCH_MANY_PLAYLISTS_FAILURE,
  FETCH_SINGLE_PLAYLIST_FAILURE,
  FETCH_SINGLE_PLAYLIST_SUCCESS,
  SET_PLAYLIST,
  REMOVE_PLAYLIST_REQUEST,
  REMOVE_PLAYLIST_FAILURE,
  REMOVE_PLAYLIST_SUCCESS,
  REMOVE_PLAYLIST_TRACK,
} from '@client/store/playlist/actionTypes';
import { PlaylistInterface } from '@client/store/playlist';

// Fetching many playlists
export const fetchPlaylistsRequest = () => ({
  type: FETCH_MANY_PLAYLISTS_REQUEST as typeof FETCH_MANY_PLAYLISTS_REQUEST,
});

export const fetchPlaylistsSuccess = (payload: PlaylistInterface[]) => ({
  payload,
  type: FETCH_MANY_PLAYLISTS_SUCCESS as typeof FETCH_MANY_PLAYLISTS_SUCCESS,
});

export const fetchPlaylistsFailure = () => ({
  type: FETCH_MANY_PLAYLISTS_FAILURE as typeof FETCH_MANY_PLAYLISTS_FAILURE,
});

// Fetching single playlist
export const fetchPlaylistRequest = (id: number) => ({
  payload: id,
  type: FETCH_SINGLE_PLAYLIST_REQUEST as typeof FETCH_SINGLE_PLAYLIST_REQUEST,
});

export const fetchPlaylistSuccess = (payload: PlaylistInterface) => ({
  payload,
  type: FETCH_SINGLE_PLAYLIST_SUCCESS as typeof FETCH_SINGLE_PLAYLIST_SUCCESS,
});

export const fetchPlaylistFailure = (id: number) => ({
  payload: id,
  type: FETCH_SINGLE_PLAYLIST_FAILURE as typeof FETCH_SINGLE_PLAYLIST_FAILURE,
});

// Delete single playlist
export const removePlaylistRequest = (id: number) => ({
  payload: id,
  type: REMOVE_PLAYLIST_REQUEST as typeof REMOVE_PLAYLIST_REQUEST,
});

export const removePlaylistSuccess = (id: number) => ({
  payload: id,
  type: REMOVE_PLAYLIST_SUCCESS as typeof REMOVE_PLAYLIST_SUCCESS,
});

export const removePlaylistFailure = (id: number) => ({
  payload: id,
  type: REMOVE_PLAYLIST_FAILURE as typeof REMOVE_PLAYLIST_FAILURE,
});

// Save single playlist
export const setSinglePlaylist = (payload: PlaylistInterface) => ({
  payload,
  type: SET_PLAYLIST as typeof SET_PLAYLIST,
});

export const removePlaylistTrack = (playlistId: number, trackId: number) => ({
  type: REMOVE_PLAYLIST_TRACK as typeof REMOVE_PLAYLIST_TRACK,
  payload: {
    playlistId,
    trackId,
  },
});

