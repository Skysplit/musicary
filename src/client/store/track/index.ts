import { fetchTracksSuccess } from '@client/store/track/actions';
import { FETCH_MANY_TRACKS_SUCCESS } from '@client/store/track/actionTypes';

export interface TrackInterface {
  id: number;
  name: string;
  sourceId: string;
  sourceType: string;
}

export interface TrackState {
  byId: {
    [key: number]: {
      isLoading: boolean;
      isRemoving: boolean;
      data: TrackInterface;
    },
  };
}

type Actions =
  | ReturnType<typeof fetchTracksSuccess>;

const initialState = {
  byId: {},
};

export default (state: TrackState = initialState, action: Actions) => {
  switch (action.type) {
    case FETCH_MANY_TRACKS_SUCCESS: {
      return {
        ...state,
        byId: action.payload.tracks.reduce(
          (accum, track) => Object.assign(accum, {
            [track.id]: {
              isLoading: false,
              isRemoving: false,
              data: track,
            },
          }),
          {},
        ),
      };
    }

    default: {
      return state;
    }
  }
};
