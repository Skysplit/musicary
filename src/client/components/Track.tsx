import { PureComponent, MouseEvent } from 'react';
import { ListItem, FontIcon } from 'react-md';
import { TrackInterface } from '@client/store/track';
import { PlaylistInterface } from '@client/store/playlist';
import client from '@client/utils/client';

type TrackProps = {
  track: TrackInterface;
  playlist: PlaylistInterface;
  active: boolean;
  onClick: (track: TrackInterface) => any;
};

export default class Track extends PureComponent<TrackProps> {
  handleClick = () => this.props.onClick(this.props.track);

  removeTrack = async (e: MouseEvent<any>) => {
    e.stopPropagation();

    const { track, playlist } = this.props;
    const confirmed = confirm(`Are you sure you want to remove ${track.name} from this playlist?`);

    if (!confirmed) {
      return;
    }

    try {
      await client.delete(`/playlists/${playlist.id}/tracks/${track.id}`);
    } catch (err) {
      // Gotta catch 'em all
    }
  }

  renderActiveIcon = () => (
    this.props.active ? <FontIcon>play_arrow</FontIcon> : null
  )

  renderRemoveIcon = () => this.props.active ? (
    null
  ) : (
    <FontIcon onClick={this.removeTrack}>
      delete
    </FontIcon>
  )

  render() {
    const { track, active } = this.props;

    return (
      <ListItem
        onClick={this.handleClick}
        primaryText={track.name}
        leftIcon={this.renderActiveIcon()}
        rightIcon={this.renderRemoveIcon()}
        active={active}
      />
    );
  }
}
