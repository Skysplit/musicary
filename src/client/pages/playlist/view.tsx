import { PureComponent, Fragment } from 'react';
import { bindActionCreators } from 'redux';
import { connect, Dispatch } from 'react-redux';
import { Request } from 'express';
import io from 'socket.io-client';
import { Snackbar } from 'react-md';
import { Context } from 'next/document';
import Error from 'next/error';
import { PlaylistInterface } from '@client/store/playlist';
import { setSinglePlaylist, removePlaylistTrack } from '@client/store/playlist/actions';
import client, { getHeaders } from '@client/utils/client';
import { fetchTracksSuccess, setSingleTrack, removeTrack } from '@client/store/track/actions';
import { TrackInterface } from '@client/store/track';
import { getUserToken } from '@client/utils/userData';
import PlaylistViewContainer from '@client/containers/PlaylistViewContainer';

interface PageContext extends Context {
  req: Request;
}

interface InitialProps {
  title: string;
  playlist?: PlaylistInterface;
  tracks?: TrackInterface[];
  errorCode?: number;
}

interface PlaylistViewPageProps extends InitialProps {
  setSinglePlaylist: typeof setSinglePlaylist;
  fetchTracksSuccess: typeof fetchTracksSuccess;
  setSingleTrack: typeof setSingleTrack;
  removePlaylistTrack: typeof removePlaylistTrack;
}

type ComponentProps = PlaylistViewPageProps & {};

interface Toast {
  text: string;
}

type ComponentState = {
  toasts: Toast[],
};

type AuthHeaders = ReturnType<typeof getHeaders>;

export class PlaylistViewPage extends PureComponent<ComponentProps, ComponentState> {
  private io: SocketIOClient.Socket;

  static async fetchPlaylist(id: number, headers: AuthHeaders) {
    const response = await client.get(`/playlists/${id}`, { headers });
    return response.data as PlaylistInterface;
  }

  static async fetchPlaylistTracks(id: number, headers: AuthHeaders) {
    const response = await client.get(`/playlists/${id}/tracks`, { headers });
    return response.data as TrackInterface[];
  }

  static async getInitialProps({ req, query }: PageContext): Promise<InitialProps> {
    let playlist: PlaylistInterface;
    let tracks: TrackInterface[];
    let errorCode: number;
    let title = 'Playlist';

    try {
      const headers = getHeaders(req);

      [playlist, tracks] = await Promise.all([
        PlaylistViewPage.fetchPlaylist(query.id as number, headers),
        PlaylistViewPage.fetchPlaylistTracks(query.id as number, headers),
      ]);

      title = `Playlist | ${playlist.name}`;
    } catch (err) {
      errorCode = err.response.status;
    }

    return {
      title,
      playlist,
      tracks,
      errorCode,
    };
  }

  state: ComponentState = {
    toasts: [],
  };

  constructor(props: ComponentProps) {
    super(props);

    if (props.playlist) {
      props.setSinglePlaylist(props.playlist);
      props.fetchTracksSuccess(props.playlist.id, props.tracks);
    }
  }

  private handleSocketConnection = () => {
    const { toasts } = this.state;
    const toast = { text: 'Connected with server!' };

    this.setState({
      toasts: [...toasts, toast],
    });
  }

  private handleSocketImportError = (playlistId: number) => {
    const { playlist } = this.props;
    const { toasts } = this.state;

    if (playlist.id !== playlistId) {
      return;
    }

    const toast = {
      text: 'There was an error while importing playlist',
    };

    this.setState({
      toasts: [...toasts, toast],
    });
  }

  private handleSocketTrack = (playlistId: number, track: TrackInterface) => {
    const { playlist } = this.props;
    const { toasts } = this.state;

    if (playlist.id !== playlistId) {
      return;
    }

    const toast = {
      text: `New track! ${track.name}`,
    };

    this.props.setSingleTrack(playlistId, track);

    this.setState({
      toasts: [...toasts, toast],
    });
  }

  private handleSocketTracksAccepted = (playlistId: number) => {
    const { playlist } = this.props;
    const { toasts } = this.state;

    if (playlist.id !== playlistId) {
      return;
    }

    const toast = {
      text: `Your tracks are being processed. They will appear on your list soon`,
    };

    this.setState({
      toasts: [...toasts, toast],
    });
  }

  private handleSocketTrackRemoved = (playlistId: number, track: TrackInterface) => {
    const { playlist } = this.props;
    const { toasts } = this.state;

    if (playlist.id !== playlistId) {
      return;
    }

    const toast = {
      text: `Track ${track.name} removed from playlist`,
    };

    this.setState({
      toasts: [...toasts, toast],
    });

    this.props.removePlaylistTrack(playlistId, track.id);
  }

  componentDidMount() {
    this.io = io({
      query: { token: getUserToken() },
      autoConnect: false,
    });

    this.io.connect();

    this.io.on('connect', this.handleSocketConnection);
    this.io.on('tracks-accepted', this.handleSocketTracksAccepted);
    this.io.on('track', this.handleSocketTrack);
    this.io.on('import-error', this.handleSocketImportError);
    this.io.on('track-removed', this.handleSocketTrackRemoved);
  }

  componentWillUnmount() {
    this.io.removeAllListeners();
    this.io.close();
  }

  handleDismiss = () => {
    const [, ...toasts] = this.state.toasts;
    this.setState({ toasts });
  }

  render() {
    const { errorCode, playlist } = this.props;

    if (errorCode) {
      return <Error statusCode={errorCode} />;
    }

    return (
      <Fragment>
        <PlaylistViewContainer playlistId={playlist.id} />
        <Snackbar
          toasts={this.state.toasts}
          onDismiss={this.handleDismiss}
          autohideTimeout={2000}
          autohide
        />
      </Fragment>
    );
  }
}

const actionCreators = {
  setSinglePlaylist,
  setSingleTrack,
  removePlaylistTrack,
  fetchTracksSuccess,
};

const mapDispatchToProps = (dispatch: Dispatch) => bindActionCreators(actionCreators, dispatch);

export default connect(null, mapDispatchToProps)(PlaylistViewPage);
