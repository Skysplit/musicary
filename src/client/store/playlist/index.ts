import { omit, map, uniq, reject } from 'lodash';
import {
  fetchPlaylistRequest,
  fetchPlaylistSuccess,
  fetchPlaylistsRequest,
  fetchPlaylistsSuccess,
  fetchPlaylistFailure,
  fetchPlaylistsFailure,
  setSinglePlaylist,
  removePlaylistRequest,
  removePlaylistSuccess,
  removePlaylistFailure,
  removePlaylistTrack,
} from '@client/store/playlist/actions';
import {
  FETCH_SINGLE_PLAYLIST_REQUEST,
  FETCH_SINGLE_PLAYLIST_SUCCESS,
  FETCH_MANY_PLAYLISTS_REQUEST,
  FETCH_MANY_PLAYLISTS_SUCCESS,
  FETCH_MANY_PLAYLISTS_FAILURE,
  FETCH_SINGLE_PLAYLIST_FAILURE,
  SET_PLAYLIST,
  REMOVE_PLAYLIST_REQUEST,
  REMOVE_PLAYLIST_SUCCESS,
  REMOVE_PLAYLIST_FAILURE,
  REMOVE_PLAYLIST_TRACK,
} from '@client/store/playlist/actionTypes';
import { fetchTracksSuccess, setSingleTrack } from '@client/store/track/actions';
import { FETCH_MANY_TRACKS_SUCCESS, SET_TRACK } from '@client/store/track/actionTypes';

export interface PlaylistInterface {
  id?: number;
  name: string;
  position?: number;
  tracks?: number[];
}

export interface PlaylistState {
  isLoading: boolean;
  byId: {
    [key: number]: {
      isLoading: boolean;
      isRemoving: boolean;
      data?: PlaylistInterface;
    };
  };
}

type Actions =
  | ReturnType<typeof fetchPlaylistRequest>
  | ReturnType<typeof fetchPlaylistSuccess>
  | ReturnType<typeof fetchPlaylistFailure>
  | ReturnType<typeof fetchPlaylistsRequest>
  | ReturnType<typeof fetchPlaylistsSuccess>
  | ReturnType<typeof fetchPlaylistsFailure>
  | ReturnType<typeof setSinglePlaylist>
  | ReturnType<typeof removePlaylistRequest>
  | ReturnType<typeof removePlaylistSuccess>
  | ReturnType<typeof removePlaylistFailure>
  | ReturnType<typeof fetchTracksSuccess>
  | ReturnType<typeof setSingleTrack>
  | ReturnType<typeof removePlaylistTrack>;


const initialState: PlaylistState = {
  isLoading: false,
  byId: {},
};

export default (state: PlaylistState = initialState, action: Actions): PlaylistState => {
  switch (action.type) {
    case FETCH_MANY_PLAYLISTS_REQUEST: {
      return {
        ...state,
        isLoading: true,
      };
    }

    case FETCH_MANY_PLAYLISTS_SUCCESS: {
      return {
        ...state,
        isLoading: false,
        byId: action.payload.reduce(
          (accum, data) => Object.assign(accum, {
            [data.id]: { data, isLoading: false },
          }),
          {},
        ),
      };
    }

    case FETCH_MANY_PLAYLISTS_FAILURE: {
      return state;
    }

    case FETCH_SINGLE_PLAYLIST_REQUEST: {
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload]: {
            isLoading: true,
            isRemoving: false,
          },
        },
      };
    }

    case FETCH_SINGLE_PLAYLIST_SUCCESS: {
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload.id]: {
            isLoading: false,
            isRemoving: false,
            data: action.payload,
          },
        },
      };
    }

    case FETCH_SINGLE_PLAYLIST_FAILURE: {
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload]: {
            isLoading: false,
            isRemoving: false,
          },
        },
      };
    }

    case REMOVE_PLAYLIST_REQUEST: {
      const item = state.byId[action.payload];
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload]: {
            ...item,
            isRemoving: true,
          },
        },
      };
    }

    case REMOVE_PLAYLIST_SUCCESS: {
      return {
        ...state,
        byId: omit(state.byId, [action.payload]),
      };
    }

    case REMOVE_PLAYLIST_FAILURE: {
      const item = state.byId[action.payload];
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload]: {
            ...item,
            isRemoving: false,
          },
        },
      };
    }

    case SET_PLAYLIST: {
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload.id]: {
            isLoading: false,
            isRemoving: false,
            data: action.payload,

          },
        },
      };
    }

    case FETCH_MANY_TRACKS_SUCCESS: {
      const { playlistId, tracks } = action.payload;
      const playlist = state.byId[playlistId];

      if (!playlist) {
        return state;
      }

      return {
        ...state,
        byId: {
          ...state.byId,
          [playlistId]: {
            ...playlist,
            data: {
              ...playlist.data,
              tracks: map(tracks, 'id'),
            },
          },
        },
      };
    }

    case SET_TRACK: {
      const { playlistId, track } = action.payload;
      const playlist = state.byId[playlistId];

      if (!playlist) {
        return;
      }

      return {
        ...state,
        byId: {
          ...state.byId,
          [playlistId]: {
            ...playlist,
            data: {
              ...playlist.data,
              tracks: uniq([
                ...playlist.data.tracks,
                track.id,
              ]),
            },
          },
        },
      };
    }

    case REMOVE_PLAYLIST_TRACK: {
      const { playlistId, trackId } = action.payload;
      const playlist = state.byId[playlistId];

      if (!playlist) {
        return;
      }

      return {
        ...state,
        byId: {
          ...state.byId,
          [playlistId]: {
            ...playlist,
            data: {
              ...playlist.data,
              tracks: reject(playlist.data.tracks, id => id === trackId),
            },
          },
        },
      };
    }

    default: {
      return state;
    }
  }
};
