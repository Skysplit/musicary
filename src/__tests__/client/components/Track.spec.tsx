import React from 'react';
import { shallow } from 'enzyme';
import Track from '@client/components/Track';
import { PlaylistInterface } from '@client/store/playlist';
import { TrackInterface } from '@client/store/track';

describe('<Track />', () => {
  const defaultProps = {
    track: { id: 1, name: 'TrackName' } as TrackInterface,
    active: true,
    playlist: { } as PlaylistInterface,
    onClick: jest.fn() as (track: TrackInterface) => void,
  };

  test('should render component', () => {
    const props = { ...defaultProps };
    const wrapper = shallow(<Track {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('when track is not active', () => {
    test('should display different component state', () => {
      const props = { ...defaultProps, active: false };
      const wrapper = shallow(<Track {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
