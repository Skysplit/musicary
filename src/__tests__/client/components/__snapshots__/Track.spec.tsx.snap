// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Track /> should render component 1`] = `
<ListItem
  active={true}
  activeClassName="md-text--theme-primary"
  animateNestedItems={true}
  component="div"
  expanderIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      keyboard_arrow_down
    </FontIcon>
  }
  itemComponent="li"
  leftIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      play_arrow
    </FontIcon>
  }
  onClick={[Function]}
  primaryText="TrackName"
  rightIcon={null}
/>
`;

exports[`<Track /> when track is not active should display different component state 1`] = `
<ListItem
  active={false}
  activeClassName="md-text--theme-primary"
  animateNestedItems={true}
  component="div"
  expanderIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      keyboard_arrow_down
    </FontIcon>
  }
  itemComponent="li"
  leftIcon={null}
  onClick={[Function]}
  primaryText="TrackName"
  rightIcon={
    <FontIcon
      iconClassName="material-icons"
      onClick={[Function]}
    >
      delete
    </FontIcon>
  }
/>
`;
